File: py/0001/main.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for i, val in enumerate(nums):
            remainder = target - val
            if remainder in cache:
                return [cache[remainder], i]
            cache[val] = i
        return [0, 0]

File: java/0001/Main.java
import java.util.*;

class Solution {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> cache = new HashMap<>();

    for (int i = 0; i < nums.length; i++) {
      int remainder = target - nums[i];
      if (cache.containsKey(remainder)) {
        return new int[]{cache.get(remainder), i};
      } else {
        cache.put(nums[i], i);
      }
    }

    throw new IllegalArgumentException("no solution");
  }
}
